src/openapi/transport/batch.ts

```diff
@@ -0,0 +1,292 @@
+import log from '../../log';
+import { shouldUseCloud } from './options';
+import type { QueueItem } from './queue';
+import TransportQueue from './queue';
+import type * as types from './types';
+
+const reUrl = /((https?:)?\/\/)?[^/]+(.*)/i;
+
+const LOG_AREA = 'TransportBatch';
+
+// -- Local methods section --
+
+type BatchResult = {
+    response: string;
+    headers?: {
+        get: (key: string) => string;
+    };
+    isNetworkError?: boolean;
+    status?: number;
+};
+
+function getParentRequestId(batchResult: BatchResult) {
+    let parentRequestId = 0;
+
+    if (batchResult.headers) {
+        parentRequestId = parseInt(batchResult.headers.get('x-request-id'), 10);
+        parentRequestId = isNaN(parentRequestId) ? 0 : parentRequestId;
+    }
+    return parentRequestId;
+}
+
+// -- Exported methods section --
+
+/**
+ * Creates a wrapper around transport to provide auto-batching functionality. If you use the default of 0ms then this transport will join
+ * together all calls that happen inside the current call stack and join them into a batch call.
+ * @class
+ * @alias saxo.openapi.TransportBatch
+ * @param {Transport} transport - Instance of the transport class to wrap.
+ * @param {string} baseUrl - Base URL for batch requests. This should be an absolute URL.
+ * @param {Object} [options]
+ * @param {number} [options.timeoutMs=0] - Timeout after starting to que items before sending a batch request.
+ * @param {string} [options.host=global.location.host] - The host to use in the batch request. If not set defaults to global.location.host.
+ * @param {Object.<string, saxo.ServiceOptions>} [options.services] - Per-service options, keyed by service path.
+ */
+
+class TransportBatch extends TransportQueue {
+    basePath = '';
+    host: null | string = null;
+    timeoutMs = 0;
+    services: types.services = {};
+    isQueueing = true;
+    nextTickTimer: ReturnType<typeof setTimeout> | boolean = false;
+
+    constructor(
+        transport: any,
+        baseUrl?: string | null,
+        options?: types.Options | null,
+    ) {
+        super(transport);
+
+        if (!baseUrl) {
+            throw new Error(
+                'Missing required parameter: baseUrl in TransportBatch',
+            );
+        }
+
+        const splitBaseUrl = baseUrl.match(reUrl);
+
+        if (!splitBaseUrl) {
+            // the regular expression will match anything but "" and "/"
+            throw new Error('baseUrl is not valid - unable to extract path');
+        }
+
+        let basePath = splitBaseUrl[3] || '/';
+
+        if (basePath[basePath.length - 1] !== '/') {
+            basePath += '/';
+        }
+
+        // Batching is a service group level facility, so isn't applicable/available for /oapi
+        this.basePath = basePath + 'openapi/';
+
+        if (options && options.host) {
+            this.host = options.host;
+        } else {
+            this.host = location.host;
+        }

```

Comment:
If we initialise at the top instead of null we don't need this else here and also the @ts-ignore in line 166 won't be needed